static/app/components/events/interfaces/spans/scrollbarManager.tsx

```diff
@@ -84,6 +87,26 @@ export class Provider extends Component<Props, State> {
     // but only for DOM elements that actually got rendered
 
     this.initializeScrollState();
+
+    const anchoredSpanHash = window.location.hash.split('#')[1];
+
+    // If the user is opening the span tree with an anchor link provided, we need to reconnect the observers after a short delay.
+    // This is because we need to wait for the window to scroll to the anchored span first, or there will be inconsistencies in
+    // the spans that are actually considered in the view. The IntersectionObserver API cannot keep up with the speed
+    // at which the window scrolls to the anchored span, and will be unable to register the spans that went out of the view
+
+    if (anchoredSpanHash) {
+      // We cannot assume the root is in view to start off, if there is an anchored span
+      this.spansInView.isRootSpanInView = false;
+      this.spanBars.forEach(spanBar => spanBar.disconnectObservers());
+
+      setTimeout(() => {
+        this.spanBars.forEach(spanBar => spanBar.connectObservers());
+      }, 500);
+      return;
+    }
```

Comment:
Setting a fixed timeout will be brittle, so I'm not sure about this. One alternative off the top of my head is doing an interval that checks for the existence of the anchored span in the viewport and then cancels itself instead to make it at least re-connect post-scroll consistently vs. randomly.

Reply:
@k-fish That's a great idea! I'll look into setting this up tomorrow